### DISCUSSION TITLE
[Getting Advice]Repeatable jobs

### DISCUSSION THREAD

**Discussion by aditodkar** [OptimalBits/bull#2262] [General]
I am trying to schedule repeatable jobs with start date and cron configuration. Lets say start date is Friday at 9PM 21st Jan and if I add cron configuration to repeat the post at every Friday. Lets say job is added on Thursday 10PM 20th Jan then the job is not executed on 21st Jan Friday 9PM but instead it is executed on 28th Friday at 9PM. Is it possible to execute job at that same time as well as repeat every Friday or any day? Any suggestions? 

My solution:
Should I add 2 separate jobs one which will get executed on Friday 9PM 21st Jan 2022 and at the same time schedule repeatable job at same time which will execute job on Jan 28th at 9PM and then every friday. 
cc: @manast

**Comment by manast** [comment]
Not sure if you did not write your example correctly, but as you wrote the job should be processed also on 21st Jan Friday 9 PM, unless I missed something...

**Reply by aditodkar** [reply]
@manast  Code: 
```
const id = uuidv4();

        const data = {
            mockdata: 'Hello world',
            user_id: 'ABC'
        }
        const jobOpts = {
            jobId: id,
            repeat: {
                startDate: new Date('Thursday Jan 20 2022 19:45 GMT+0530 (India Standard Time)'),
                cron: "45 19 * * 4"
            }
        }
        await postsQueue.add(data, jobOpts);
```
Please check above code it does not execute job on thursday Jan 20 2022 19:45 any idea why so? am I missing something? In bull arena I see in delayed jobs next execution is on 27th Thursday 2022 at 19:45 why not on Thursday Jan 20th at 19:45??

![Screenshot_4](https://user-images.githubusercontent.com/16613832/150355520-e8748e8a-3ad4-4118-b710-54e2f0d49ab6.png)

**Comment by aditodkar** [comment]
Sorry my bad. I understood that if startDate is used with cron configuration then it will not execute the job at that specific time. 
Example 1:
startDate: Thursday Jan 20 2022 19:45 GMT+0530 (India Standard Time)
cron: "45 19 * * 4"
Above example it will execute job on 27th Jan 2022 and not on Jan 20th thursday.

Example 2:
cron: "45 19 * * 4" this will execute job on Jan 20 2022 19:45 thursday as well as execute it on 27th Jan 2022 19:45. Note: here we don't use startDate.

**Reply by manast** [reply]
We are just passing the opts to cron-parser, so I think it should work, but maybe there is some timezone issue? you could test with cron-parser to see if the next date you get is what you expect: https://www.npmjs.com/package/cron-parser
